Feature/UI styling

RileyManda opened the pull request:
I have completed the task styling the application according to the  stipulated  [requirements](https://github.com/microverseinc/curriculum-react-redux/blob/main/bookstore/styling.md): 🌱 

**The following implementations were made:** 🌱 

- [x] Added any missing JSX elements to project (without adding extra functionality)
- [x] Styled the application according to the  elements in the Zeplin design
- [x] Used CSS rules detailed in Zeplin to style the website.

In addition, I have ensured that this project fulfills the following default requirements :
✅ - The application behaves the same way after the refactoring
✅ - There are [no linter errors](https://github.com/microverseinc/linters-config).
✅ - Used correct flow [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).
✅ - Documented my work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).
✅ - Followed the list of the [best practices for HTML & CSS](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/html_css_best_practices.md) and  [JavaScript](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md).


Please review my implementation and suggest any improvements and changes.

Thank you 🙏

RileyManda committed:
Implemented logo color and app font

RileyManda committed:
Added font

RileyManda committed:
Implemented profile icon

RileyManda committed:
Implemented card and card height and alignment

RileyManda committed:
Book item padding and color

RileyManda committed:
Implemented background color

RileyManda committed:
Button padding,width and color

RileyManda committed:
Remove button alignment

RileyManda committed:
Implemented update progress btn color

RileyManda committed:
Implemented button group

RileyManda committed:
Realigned button group

RileyManda committed:
Book form styling

RileyManda committed:
Implemented progress and progres alignment

RileyManda committed:
Updated center elements alignment

RileyManda committed:
Added percentages to book array

RileyManda committed:
Updated margin top

RileyManda committed:
Updated completed element margin

Munsa1 commented on src/App.css:
```diff
@@ -49,28 +78,89 @@
+/* main container */
+
 .container {
   display: flex;
```
- [ ] Please use reusable classes for reoccurring styles. For example `display: flex:`  is repeated 18 times in this file. It would be cleaner to create a class in CSS with the style and then use that className on all elements that are affected by it in the components. Kindly do so.

e.g
```
.d-flex {
  display:flex;
}
```

```
<div className="d-flex">
</div>

RileyManda replied:
Thank you for reviewing my implementation @Munsa1. I will work on the suggested changes. 🙏

Munsa1 commented on src/index.css:
```diff
@@ -11,7 +11,8 @@ body {
     'Fira Sans',
     'Droid Sans',
     'Helvetica Neue',
-    sans-serif;
+    sans-serif,
+    RobotoSlab;
```
- [ ] The current way in which you have added the `font-family` property is a top-down approach, where if the first font is available, it is applied throughout the file, if not the next font is applied if available, and so on. This means the whole app has one font family, but this is not the case with the given design. We see that some elements have the `Montserrat` font while others have the `RobotoSlab`. Kindly update your styles to match the given design

Given design
![image](https://github.com/RileyManda/bookstore/assets/81932478/a05e343b-ee05-4725-abae-403d2fdb14e6)

Your design
![image](https://github.com/RileyManda/bookstore/assets/81932478/79eb50ec-7f50-42ae-bbbb-6aca1759cfcd)

- The given design uses `RobotoSlab` for all book card components except the `Completed` element which uses `Montserrat`, kindly fix this.

### Kindly use reusable classes for best practices.

RileyManda replied:
Noted with thanks @Munsa1 🙏 .I will work on the suggested changes,Thank you.

Munsa1 commented on src/App.css:
```diff
@@ -49,28 +78,89 @@
+}
+
+.person-icon {
+  align-items: center;
```
- [x] The same applies to the `align-items: center;` style. It is repeated 14 times. It would be more efficient to create a class and then reuse that className where it applies to the elements. Check the first comment above for an example

Munsa1 commented on src/App.css:
```diff
@@ -85,33 +175,189 @@
+  padding: 0.438rem 1.188rem 0.5rem 1.375rem;
+  border-radius: 3px;
+  border: none;
+  color: var(--white-background-theme);
```
- [x] Again here a reusable class would be the best way to use this style in our CSS as it is repeated 6 times. For more information refer to the first comment on how to apply it.

Munsa1 reviewed (changes requested):
Hi @RileyManda , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
Added All Elements In The Design :heavy_check_mark: 
No Inline Styling :heavy_check_mark: 
Descriptive ReadMe File :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._
### Due to insufficient files changed I will leave a comment here
- [x] Kindly fix the link to your Twitter account in the readme file under the Author section as it leads to a profile that does not exist.
![image](https://github.com/RileyManda/bookstore/assets/81932478/7e7672db-b39b-480a-9fa2-f4f6666f73e2)

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- n/a

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

RileyManda committed:
Fixed broken twitter account lunk

RileyManda committed:
Created reusable classes for flex and align-items

RileyManda committed:
Implemented center align on form wrapper

RileyManda committed:
Resolved repeating align-items-center class

RileyManda committed:
Updated flex to reusable css class

RileyManda committed:
Implemented reusable flex container class

RileyManda committed:
Resolved repeating border none class

RileyManda committed:
Resolved repeating margin-padding in navbar style

RileyManda committed:
removed outline on input

RileyManda committed:
Resolved repeating css classes

RileyManda committed:
Updated fonts for all elements

RileyManda committed:
Resolved repeating font face

RileyManda committed:
Add button font face

RileyManda committed:
Update vl line color

RileyManda committed:
Add new book font face

RileyManda commented:
I have completed the performing iterations on the project UI as requested by the previous reviewer: 🌱 

**The following iterations were made:** 🌱 
- [x]removed repeating CSS classes
- [x] Created reusable fonts and style classes
- [x] Implemented the correct fonts for each section(**Montserrat and RobotoSlab**)


In addition, I have ensured that this project fulfills the following default requirements :
✅ - The application behaves the same way after the refactoring
✅ - There are [no linter errors](https://github.com/microverseinc/linters-config).
✅ - Used correct flow [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).
✅ - Documented my work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).
✅ - Followed the list of the [best practices for HTML & CSS](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/html_css_best_practices.md) and  [JavaScript](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md).


Please review my implementation and suggest any improvements and changes.

Thank you 🙏

RileyManda committed:
Removed percentage in progress circle center to match design

RileyManda commented:
I have completed the performing iterations on the project UI as requested by the previous reviewer: 🌱 

**The following iterations were made:** 🌱 
- [x] Updated twitter link in readme
- [x] Removed repeating CSS classes
- [x] Created reusable fonts and style classes
- [x] Implemented the correct fonts for each section(**Montserrat and RobotoSlab**)
- [x] Removed percentage in the circular progress to match the zeplin design.


In addition, I have ensured that this project fulfills the following default requirements :
✅ - The application behaves the same way after the refactoring
✅ - There are [no linter errors](https://github.com/microverseinc/linters-config).
✅ - Used correct flow [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).
✅ - Documented my work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).
✅ - Followed the list of the [best practices for HTML & CSS](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/html_css_best_practices.md) and  [JavaScript](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md).


Please review my implementation and suggest any improvements and changes.

Thank you 🙏

RileyManda committed:
Updated font size

RileyManda commented:
I have completed the performing iterations on the project UI as requested by the previous reviewer: 🌱 

**The following iterations were made:** 🌱 
- [x] Updated twitter link in readme
- [x] Removed repeating CSS classes
- [x] Created reusable fonts and style classes
- [x] Implemented the correct fonts for each section(**Montserrat and RobotoSlab**)
- [x] Removed percentage in the circular progress to match the zeplin design.


In addition, I have ensured that this project fulfills the following default requirements :
✅ - The application behaves the same way after the refactoring
✅ - There are [no linter errors](https://github.com/microverseinc/linters-config).
✅ - Used correct flow [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).
✅ - Documented my work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).
✅ - Followed the list of the [best practices for HTML & CSS](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/html_css_best_practices.md) and  [JavaScript](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md).


Please review my implementation and suggest any improvements and changes.

Thank you 🙏

julie-ify reviewed (approved):
Hi @RileyManda,

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️ 

Cheers and Happy coding!👏👏👏

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
